File: app/src/main/java/com/anokmik/tripassistant/util/DateUtils.java
package com.anokmik.tripassistant.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public final class DateUtils {

    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy", Locale.US);
    private static final String FORMAT = "%s - %s";

    private DateUtils() {

    }

    public static boolean validDates(long startDate, long finishDate) {
        Calendar calendar = Calendar.getInstance();

        calendar.setTimeInMillis(startDate);
        int startYear = calendar.get(Calendar.YEAR);
        int startMonth = calendar.get(Calendar.MONTH);
        int startDayOfMonth = calendar.get(Calendar.DAY_OF_MONTH);

        calendar.setTimeInMillis(finishDate);
        int finishYear = calendar.get(Calendar.YEAR);
        int finishMonth = calendar.get(Calendar.MONTH);
        int finishDayOfMonth = calendar.get(Calendar.DAY_OF_MONTH);

        return (finishYear >= startYear) && (finishMonth >= startMonth) && (finishDayOfMonth >= startDayOfMonth);
    }

    public static String format(long startDate, long finishDate) {
        String startDateValue = toString(startDate);
        String finishDateValue = toString(finishDate);
        return startDateValue.equals(finishDateValue)
                ? startDateValue : String.format(FORMAT, startDateValue, finishDateValue);
    }

    public static String toString(long time) {
        return DATE_FORMAT.format(new Date(time));
    }

    public static long toTime(String date) {
        try {
            return DATE_FORMAT.parse(date).getTime();
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return 0;
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/util/DateUtils.kt
package com.anokmik.tripassistant.util

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    private val DATE_FORMAT = SimpleDateFormat("dd/MM/yyyy", Locale.US)
    private val FORMAT = "%s - %s"

    @JvmStatic
    fun validDates(startDate: Long?, finishDate: Long?): Boolean {
        if (startDate == null || finishDate == null) {
            return false
        }

        val calendar = Calendar.getInstance()

        calendar.timeInMillis = startDate
        val startYear = calendar.get(Calendar.YEAR)
        val startMonth = calendar.get(Calendar.MONTH)
        val startDayOfMonth = calendar.get(Calendar.DAY_OF_MONTH)

        calendar.timeInMillis = finishDate
        val finishYear = calendar.get(Calendar.YEAR)
        val finishMonth = calendar.get(Calendar.MONTH)
        val finishDayOfMonth = calendar.get(Calendar.DAY_OF_MONTH)

        return finishYear >= startYear && finishMonth >= startMonth && finishDayOfMonth >= startDayOfMonth
    }

    @JvmStatic
    fun format(startDate: Long, finishDate: Long): String {
        val startDateValue = toString(startDate)
        val finishDateValue = toString(finishDate)
        return if (startDateValue.equals(finishDateValue)) startDateValue else String.format(FORMAT, startDateValue, finishDateValue)
    }

    @JvmStatic
    fun toString(time: Long) = DATE_FORMAT.format(Date(time))

    @JvmStatic
    fun toTime(date: String): Long {
        try {
            return DATE_FORMAT.parse(date).time
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        return 0
    }

}